pom.xml

```diff
@@ -151,11 +151,36 @@
       <plugin>
         <artifactId>maven-enforcer-plugin</artifactId>
         <executions>
+          <execution>
+            <id>ban-javax-servlet-api</id>
+            <goals>
+              <goal>enforce</goal>
+            </goals>
+            <phase>validate</phase>
+            <configuration>
+              <rules>
+                <bannedDependencies>
+                  <excludes>
+                    <exclude>javax.servlet:*</exclude>
```

Comment:
I don't understand: the name of this enforcement is `ban-javax-servlet-api`, but pretty much every permutation of `javax.servlet.*` is excluded?

Reply:
That is a maven coordinate ban, not a java package ban.
Notice the `:`?  
The format of that line is `groupId[:artifactId][:version][:type][:scope][:classifier]`
So this usage is just banning the maven groupId of `javax.servlet` with any artifact/version/type etc ...

Reply:
Ah, I see the confusion: I thought the `<exclude>` was enumerating groupId.artifactIds to _exclude_ from the ban, whereas you're saying they are enumerating things to _include_!

Reply:
Per the rule: the `<exclude>` list is what is banned, the `<include>` list are exceptions to what is banned.